crates/bevy_render/src/primitives/mod.rs

```diff
@@ -192,6 +192,18 @@ impl CubemapFrusta {
     }
 }
 
+#[derive(Clone, Copy, Debug, Default)]
+pub struct Line {
```

Comment:
I think we should provide the result as a pair of Vec3's, and not a `Line` struct. 

1. Based on my previous contributions, cart's preference has been to not wrap in geometric structs
2. We have a geometric primitive RFC that this may conflict with.

Reply:
I have very similar code to what's here, and I use a `Ray<Vec3>`:

```Rust
pub struct Ray<V> {
    pub origin: V,
    pub direction: V,
}
```

A ray is useful, because the math is simpler for intersections, and the equation is `self.origin + t * self.direction`. I use them often. It's also useful to implement `Mul<Ray<Vec3>, Output=Ray<Vec3>>` for transforms.

Reply:
The [Shapes RFC](https://github.com/bevyengine/rfcs/pull/12) also has rays.

Reply:
Agreed a ray is more apt here. My opinion is we defer adding primitives to the RFC impl, especially considering we don't have cart's final blessing.